samples/performance/image_compression_control/image_compression_control.cpp

```diff
@@ -0,0 +1,587 @@
+
+		compression_control_features.imageCompressionControl = VK_TRUE;
+	}
+
+	if (gpu.is_extension_supported(VK_EXT_IMAGE_COMPRESSION_CONTROL_SWAPCHAIN_EXTENSION_NAME))
+	{
+		auto &compression_control_swapchain_features = gpu.request_extension_features<VkPhysicalDeviceImageCompressionControlSwapchainFeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_COMPRESSION_CONTROL_SWAPCHAIN_FEATURES_EXT);
+
+		compression_control_swapchain_features.imageCompressionControlSwapchain = VK_TRUE;
+	}
+}
+
+bool ImageCompressionControlSample::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!VulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	load_scene("scenes/sponza/Sponza01.gltf");
+
+	auto &camera_node = vkb::add_free_camera(get_scene(), "main_camera", get_render_context().get_surface_extent());
+	camera            = dynamic_cast<vkb::sg::PerspectiveCamera *>(&camera_node.get_component<vkb::sg::Camera>());
+
+	vkb::ShaderSource scene_vs("base.vert");
+	vkb::ShaderSource scene_fs("base.frag");
+	auto              scene_subpass = std::make_unique<vkb::ForwardSubpass>(get_render_context(), std::move(scene_vs), std::move(scene_fs), get_scene(), *camera);
+	scene_subpass->set_output_attachments({(int) Attachments::Color});
+
+	// Forward rendering pass
+	auto render_pipeline = std::make_unique<vkb::RenderPipeline>();
+	render_pipeline->add_subpass(std::move(scene_subpass));
+	render_pipeline->set_load_store(scene_load_store);
+	set_render_pipeline(std::move(render_pipeline));
+
+	// Post-processing pass (chromatic aberration)
+	vkb::ShaderSource postprocessing_vs("postprocessing/postprocessing.vert");
+	postprocessing_pipeline = std::make_unique<vkb::PostProcessingPipeline>(get_render_context(), std::move(postprocessing_vs));
+	postprocessing_pipeline->add_pass().add_subpass(vkb::ShaderSource("postprocessing/chromatic_aberration.frag"));
+
+	// Trigger recreation of Swapchain and render targets, with initial compression parameters
+	update_render_targets();
+
+	get_stats().request_stats({vkb::StatIndex::frame_times,
+	                           vkb::StatIndex::gpu_ext_write_bytes});
+
+	create_gui(*window, &get_stats());
+
+	// Hide GUI compression options other than default if the required extension is not supported
+	if (!get_device().is_enabled(VK_EXT_IMAGE_COMPRESSION_CONTROL_EXTENSION_NAME))
+	{
+		for (int i = 0; i < (int) TargetCompression::Count; i++)
+		{
+			if (static_cast<TargetCompression>(i) != TargetCompression::Default)
+			{
+				gui_skip_compression_values.insert(static_cast<TargetCompression>(i));
+			}
+		}
+	}
+
+	return true;
+}
+
+void ImageCompressionControlSample::create_render_context()
+{
+	/**
+	 * The framework expects a prioritized list of surface formats. For this sample, include
+	 * only those that can be compressed.
+	 */
+	std::vector<vk::SurfaceFormatKHR> surface_formats_that_support_compression;
```

Comment:
You should not use data types from the C++-bindings of vulkan in a C-bindings-sample

Reply:
I had to do this after the rebase, the new header does not provide an alternative, does it?

Reply:
OK I see we have

```
	void create_render_context(const std::vector<SurfaceFormatType> &surface_priority_list);
```

but maybe there's a bug, for some reason when building I was getting errors because the function sees a C++ binding. Might be related to the issue with the AFBC sample that crashes now when calling `update_swapchain`?

Reply:
OK ignore my previous comment, I think I implemented what you suggested in my last commit, thanks!